### ISSUE TITLE
Main Thread Checker warning UAEventManager (v8.6.3)

### ISSUE THREAD

**Issue by kpcwatson** [urbanairship/ios-library#160]
We are receiving the following runtime warning in version 8.6.3 of the SDK.

![image](https://user-images.githubusercontent.com/510515/33647062-fade5aec-da20-11e7-98ad-51b3d93e3111.png)

```
2017-12-06 00:59:28.834910-0500 APP[18488:411251] [D] -[UAEventManager scheduleUploadWithDelay:] [Line 269] Enqueuing attempt to schedule event upload with delay on main queue.
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIApplication applicationState]
PID: 18488, TID: 411633, Thread name: (none), Queue name: com.apple.root.background-qos, QoS: 9
Backtrace:
4   AirshipKit                          0x000000010b2e2c4c -[UAEventManager scheduleUpload] + 76
5   AirshipKit                          0x000000010b2e2941 -[UAEventManager addEvent:sessionID:] + 273
6   AirshipKit                          0x000000010b2b68bc -[UAAnalytics addEvent:] + 652
7   AirshipKit                          0x000000010b2b70ef -[UAAnalytics associateDeviceIdentifiers:] + 255
8   APP                          2017-12-06 00:59:28.845200-0500 APP[18488:411668] Could not successfully update network info during initialization.
       0x0000000102e134f4 _T03APP11AppDelegateC17setupUrbanAirship33_03F2124390C7556F66B60AA233713CD7LLyyFyycfU_ + 1044
9   APP                                 0x0000000102419e69 _T0Ix_IyB_TR + 41
10  libdispatch.dylib                   0x000000010f5153f7 _dispatch_call_block_and_release + 12
11  libdispatch.dylib                   0x000000010f51643c _dispatch_client_callout + 8
12  libdispatch.dylib                   0x000000010f522499 _dispatch_root_queue_drain + 1444
13  libdispatch.dylib                   0x000000010f521e97 _dispatch_worker_thread3 + 132
14  libsystem_pthread.dylib             0x000000010f9de1ca _pthread_wqthread + 1387
15  libsystem_pthread.dylib             0x000000010f9ddc4d start_wqthread + 13
2017-12-06 00:59:28.873190-0500 APP[18488:411633] [reports] Main Thread Checker: UI API called on a background thread: -[UIApplication applicationState]
PID: 18488, TID: 411633, Thread name: (none), Queue name: com.apple.root.background-qos, QoS: 9
Backtrace:
4   AirshipKit                          0x000000010b2e2c4c -[UAEventManager scheduleUpload] + 76
5   AirshipKit                          0x000000010b2e2941 -[UAEventManager addEvent:sessionID:] + 273
6   AirshipKit                          0x000000010b2b68bc -[UAAnalytics addEvent:] + 652
7   AirshipKit                          0x000000010b2b70ef -[UAAnalytics associateDeviceIdentifiers:] + 255
8   APP                                 0x0000000102e134f4 _T03APP11AppDelegateC17setupUrbanAirship33_03F2124390C7556F66B60AA233713CD7LLyyFyycfU_ + 1044
9   APP                                 0x0000000102419e69 _T0Ix_IyB_TR + 41
10  libdispatch.dylib                   0x000000010f5153f7 _dispatch_call_block_and_release + 12
11  libdispatch.dylib                   0x000000010f51643c _dispatch_client_callout + 8
12  libdispatch.dylib                   0x000000010f522499 _dispatch_root_queue_drain + 1444
13  libdispatch.dylib                   0x000000010f521e97 _dispatch_worker_thread3 + 132
14  libsystem_pthread.dylib             0x000000010f9de1ca _pthread_wqthread + 1387
15  libsystem_pthread.dylib             0x000000010f9ddc4d start_wqthread + 13
2017-12-06 00:59:29.426147-0500 APP[18488:411633] [D] -[UAEventManager scheduleUploadWithDelay:] [Line 269] Enqueuing attempt to schedule event upload with delay on main queue.
```

**Comment by rlepinski** [comment]
Thanks for the report. It looks like the call is originating from your setup airship method. For now, please dispatch editing associated identifiers onto the main thread. In our next SDK version we will address this issue.

**Comment by rlepinski** [comment]
Fixed in 9.0.1
